tools/build_api.py

```diff
@@ -426,20 +430,33 @@ def merge_region_list(region_list, destination, notify, padding=b'\xFF'):
             notify.info("  Filling region %s with %s" % (region.name, region.filename))
             part = intelhex_offset(region.filename, offset=region.start)
             part.start_addr = None
-            part_size = (part.maxaddr() - part.minaddr()) + 1
+
+            # make same assumption as in region builder; first segment must fit.
+            # this is only to get a neat ToolException anyway, since IntelHex.merge will
+            # throw intelhex.AddressOverlapError if there's overlapping
+            part_size = 0
+            for es in part.segments():
+                # Add padding in between segments starting from end of first segment
+                if (len(part.segments()) > 1 and (merged.maxaddr() != None)):
+                    pad_size = es[0] - (merged.maxaddr() + 1)
+                    merged.puts(merged.maxaddr()+1, padding * pad_size)
```

Comment:
Please add spaces around `+` for consistency with the rest of the code base. Further, this should probably check to see if we _need_ padding, and this breaks when you have an image that's before the active region and after it, as it will add padding that overlaps with the active region.